### ISSUE TITLE
Uncaught ReferenceError: process is not defined

### ISSUE THREAD

**Issue by tvld** [elreco/vue-tailwind-datepicker#58]
This is quite terrible. The package uses a @babel that tries to access process.env That will break on some browsers that need transpile, without warning. Disaster))

But that is not longer allowed. You should use import.meta.env 

Ok, lost half a day debugging software and moving on to new date picker. A shame as the UI was flawless and the best I have seen so far...

**Comment by elreco** [comment]
I don't understand your issue, we don't have any process.env I don't know where you find it

**Comment by tvld** [comment]
It is there; you pull it in in Babel. You can grep contents for 'process.env' in node_modules folder...

 Maybe it is an old dependency of an old version of date-fns ?  Just a hunch...

**Comment by tvld** [comment]
Same issue in v1.2.10. As soon as I include the package as GlobalComponent, it chokes:

```ts
// Register all global components

import tDatePicker from './lib/form/tDatePicker.vue'

export default (app: App): void => {
  app
   .component('tDatePicker', tDatePicker)
}

declare module '@vue/runtime-core' {
  export interface GlobalComponents {
    tDatePicker: typeof tDatePicker
  }
}
```
and the component itself:
```ts
<!-- Date Picker with modal

Using: 
  https://vue-tailwind-datepicker.com/advanced-features.html
  https://date-fns.org/

Playground:
  https://stackblitz.com/edit/vue-tailwind-datepicker-97rn7o?file=tailwind.config.js,src%2Fcomponents%2FPlayground.vue

-->

<template>
  <t-modal-dialog narrow header-class="bg-secondary text-white">
    <template #header>{{ label }}</template>

    <template #show>
      <div class="flex items-center space-x-1 cursor-pointer">
        <t-icon :size="20" icon="calendar_today" color="info" btn />
        <t-date :date="modelValue" :placeholder="placeholder" />
        <slot />
      </div>
    </template>

    <template #default="{ toggleModal }">
      <vue-tailwind-datepicker
        v-model="dateValue"
        :i18n="$settings.locale"
        :formatter="{
          date: `YYYY-MM-DD HH:mm:ss`,
          month: 'MMM'
        }"
        no-input
        as-single
        :auto-apply="false"
        :disable-date="disableDate"
        @update:model-value="dateChanged($event, toggleModal)"
      />
    </template>

    <template v-if="!!$slots.footer" #footer="{ toggleModal }">
      <slot name="footer" :toggle-modal="toggleModal" />
    </template>
  </t-modal-dialog>
</template>

<script setup lang="ts">
  import type { PropType } from 'vue'
  import { format, compareAsc, parseISO, parse } from 'date-fns'

  import VueTailwindDatepicker from 'vue-tailwind-datepicker'

  import logger from '@/lib/logger'
  import { enumBooleanBody } from '@babel/types'

  const props = defineProps({
    modelValue: { type: String as PropType<NullableTDateISO> },

    // @TODO:
    // disabled: { type: Boolean, default: false }, // all visible dimmed; no click
    // readonly: { type: Boolean, default: false }, // date visible listed only; no click icon
    closeOnApply: { type: Boolean, default: false },

    label: { type: String, default: 'Select a date' }, // header
    placeholder: { type: String, default: '…' },

    disableDate: { type: Function as PropType<(date: Date) => boolean>, default: () => false },
    errorMsg: { type: String, default: '' } // error to show; for example startdate > end date
  })

  const $settings = settingsStore()

  const emit = defineEmits(['update:modelValue', 'changed'])

  // Tailwind datepicker has Typescript bug with normal date; so we need to use object or string
  //   https://github.com/elreco/vue-tailwind-datepicker/issues/46

  const twPickerFormat = 'yyyy-MM-dd kk:mm:ss'
  const dateValue = computed({
    get: (): string => {
      const dateIsoStr = props.modelValue === null ? new Date().toISOString() : props.modelValue
      return format(new Date(dateIsoStr as string), twPickerFormat)
    },
    set(newDate: string) {
      const newIsoValue = parse(newDate, twPickerFormat, new Date()).toISOString()
      emit('update:modelValue', newIsoValue)
      emit('changed', newIsoValue)
    }
  })

  const dateChanged = (newDate: any, toggleModal: Function) => {
    if (props.closeOnApply) {
      toggleModal()
    }
  }
</script>

```

![Screenshot from 2023-01-26 15-53-17](https://user-images.githubusercontent.com/986182/214868024-94a06a8a-a0fe-4687-820b-b70bb9cd0fb9.png)
![Screenshot from 2023-01-26 15-53-31](https://user-images.githubusercontent.com/986182/214868033-708f2e92-1a07-4175-9d66-e7e4ce7b2054.png)

**Comment by tvld** [comment]
Ok, found it. I went through all commit lines one by one and found that vs-code had automatically tried to solve the typing errors by including a babel/type import. It did not help, but the import was not removed thereafter... So, when I did the error went away.
Still, process.env should not have been used... but it was an old package included as a deep dependency...

anyway, I close. Sorry for the inconvenience...
